tests/integration/test_split.py

```diff
@@ -0,0 +1,310 @@
+        expected_exc = ValueError
+        func_num = getattr(num, func_name)
+        func_np = getattr(np, func_name)
+        with pytest.raises(expected_exc):
+            func_np(None, 1)
+        with pytest.raises(expected_exc):
+            func_num(None, 1)
+
+    @pytest.mark.parametrize("indices", (-2, 0, "hi", 1.0, None))
+    @pytest.mark.parametrize("func_name", ARG_FUNCS)
+    def test_indices_negative_different_split(self, func_name, indices):
+        ary = num.arange(10)
+        func_num = getattr(num, func_name)
+        func_np = getattr(np, func_name)
+
+        expected_exc = ValueError
+        with pytest.raises(expected_exc):
+            func_num(ary, indices)
+        with pytest.raises(expected_exc):
+            func_np(ary, indices)
+
+    @pytest.mark.xfail
+    def test_dimensions_vsplit(self):
+        ary = []
+        expected_exc = ValueError
+        with pytest.raises(expected_exc):
+            num.vsplit(ary, 1)
+            # cuNumeric returns [array([], dtype=float64)]
+        with pytest.raises(expected_exc):
+            np.vsplit(ary, 1)
+            # Numpy raises
+            # ValueError: vsplit only works on arrays of 2 or more dimensions
+
+    @pytest.mark.xfail
+    def test_dimensions_vsplit_1(self):
+        ary = np.random.randint(low=0, high=100, size=(5))
+        expected_exc = ValueError
+        with pytest.raises(expected_exc):
+            num.vsplit(ary, 1)
+            # cuNumeric returns the array
+        with pytest.raises(expected_exc):
+            np.vsplit(ary, 1)
+            # Numpy raises
+            # ValueError: vsplit only works on arrays of 2 or more dimensions
+
+    @pytest.mark.xfail
+    def test_dimensions_hsplit_0(self):
+        ary = np.random.randint(low=0, high=100, size=(0,))
+        expected_exc = ValueError
+        with pytest.raises(expected_exc):
+            num.hsplit(ary, 1)
+        with pytest.raises(expected_exc):
+            np.hsplit(ary, 1)
+            # Numpy returns  [array([], dtype=int64)]
+
+    def test_dimensions_hsplit_1(self):
+        ary = num.arange(10)
+        expected_exc = ValueError
+        with pytest.raises(expected_exc):
+            num.hsplit(ary, 1)
+        with pytest.raises(expected_exc):
+            np.hsplit(ary, 1)
+
+    @pytest.mark.parametrize("size", ((0,), (10,), (5, 5)))
+    def test_dimensions_dsplit(self, size):
+        ary = np.random.randint(low=0, high=100, size=size)
+        expected_exc = ValueError
+        with pytest.raises(expected_exc):
+            num.dsplit(ary, 1)
+        with pytest.raises(expected_exc):
+            np.dsplit(ary, 1)
+
+
+def compare_array(a, b):
+    """
+    Compare two array using zip method.
+    """
+    if len(a) != len(b):
+        return False
+    else:
+        for each in zip(a, b):
+            if not np.array_equal(*each):
+                return False
+    return True
+
+
+def get_indices(arr, axis):
```

Comment:
A short docstring that describes a little about the range of inputs/outputs, what this function actually does would be helpful since it is fairly complicated

Reply:
fix